finetuner/tuner/paddle/losses.py

```diff
@@ -187,3 +187,48 @@ def get_default_miner(
             return TripletMiner()
         else:
             return TripletSessionMiner()
+
+
+class NTXentLoss(nn.Layer, BaseLoss[paddle.Tensor]):
+    """Compute the NTXent (Normalized Temeprature Cross-Entropy) loss.
+
+    This loss function is a temperature-adjusted cross-entropy loss, as defined in the
+    `SimCLR paper <https://arxiv.org/abs/2002.05709>`. It operates on batches where
+    there are two views of each instance
+    """
+
+    def __init__(self, temperature: float = 0.5) -> None:
+        """Initialize the loss instance.
+
+        :param temerature: The temperature parameter
+        """
+        super().__init__()
+
+        self.temperature = temperature
+
+    def forward(
+        self, embeddings: paddle.Tensor, labels: paddle.Tensor
+    ) -> paddle.Tensor:
+        """Compute the loss.
+
+        :param embeddings: An ``[N, d]`` tensor of embeddings.
+        :param labels: An ``[N,]`` tensor of item labels. It is expected that each label
+            appears two times.
+        """
+        assert embeddings.shape[0] == labels.shape[0]
+
+        sim = (1 - get_distance(embeddings, 'cosine')) / self.temperature
+        diag = paddle.eye(sim.shape[0], dtype=sim.dtype)
+        diag = paddle.to_tensor(diag, place=sim.place)
+        labels1, labels2 = labels.unsqueeze(1), labels.unsqueeze(0)
+
+        pos_samples = paddle.cast(labels1 == labels2, sim.dtype) - diag
+
+        if not (pos_samples.sum(axis=1) == 1).all().item():
+            raise ValueError('There need to be two views of each label in the batch.')
```

Comment:
i think it dependents on the `views_per_instance`? this need to change from:
> There need to be two views of each label in the batch.
to
>There need to be at least two views of each label in the batch.

Reply:
Right now this loss only works properly with exactly two views (that's how it was designed originally). We can extend it to more than two views (PML does it), but I think the original implementation is good for now